client/my-sites/site-settings/jetpack-credentials/index.jsx

```diff
@@ -14,47 +13,37 @@ import CompactCard from 'components/card/compact';
 import CredentialsSetupFlow from './credentials-setup-flow';
 import CredentialsConfigured from './credentials-configured';
 import Gridicon from 'gridicons';
-import QueryRewindStatus from 'components/data/query-rewind-status';
-import QueryJetpackCredentials from 'components/data/query-jetpack-credentials';
+import QueryRewindState from 'components/data/query-rewind-state';
 import { getSelectedSiteId } from 'state/ui/selectors';
-import { hasMainCredentials, isRewindActive } from 'state/selectors';
+import { getRewindState } from 'state/selectors';
 
 class Backups extends Component {
-	static propTypes = {
-		hasMainCredentials: PropTypes.bool,
-		isRewindActive: PropTypes.bool,
-		siteId: PropTypes.number.isRequired,
-	};
-
 	render() {
-		const {
-			hasMainCredentials, // eslint-disable-line no-shadow
-			isRewindActive, // eslint-disable-line no-shadow
-			translate,
-			siteId,
-		} = this.props;
+		const { rewindState, siteId, translate } = this.props;
+		const hasAuthorized = rewindState.state === 'provisioning' || rewindState.state === 'active';
```

Comment:
Should we consider having some utilities to standardize state checks? We're doing more than a few `rewindState.reason === 'vp_active_on_site'`, `rewindState.state === 'provisioning'`, `rewindState.state === 'active'` and others.

Reply:
this is something I've spoken with @rcoll about and I have mixed feelings. pulling in @gwwar and @samouri on this for their thoughts.

in short, I feel that in this case where we have well-documented data types and semantics for these values that I'm comfortable leaving the abstraction at the `rewindState` object. so far I haven't seen too many actual duplicate checks against the same information. that is, maybe in a  couple places we are checking `rewindState.state === 'provisioning' || rewindState.state === 'active'` but not enough to make me think it's ripe for turning into a function.

I'm biased here against the project-wide problems we have with selector bloat and module/import bloat and the uniqueness of having such a well-defined type…

for instance, I'm not persuaded that we'd gain anything by having `hasAuthorized( rewindState )` in three places vs. `rewindState.state === 'provisioning' || rewindState.state === 'active'` in those same three places.

definitely open to your thoughts.

Reply:
What reads poorly is the `rewindState.state`. This can probably be simplified down to a string const `rewindStatus` in mapStateToProps. I would avoid a selector in this case if the enum for state is well defined.

Reply:
Kerry is right: combining the object.prop path in a const would help the readability.